stacks-signer/src/runloop.rs

```diff
@@ -649,14 +709,46 @@ impl<C: Coordinator> RunLoop<C> {
                         }
                         EpochId::Epoch25 => {
                             debug!("Received a DKG result, but are in epoch 2.5. Broadcast the transaction to the mempool.");
-                            //TODO: Cast the aggregate public key vote here
+                            match self
+                                .stacks_client
+                                .cast_vote_for_aggregate_public_key(point.clone(), 0)
+                            {
+                                Ok(txid) => {
+                                    self.transactions.push(txid);
+                                    println!(
+                                        "Successfully cast aggregate public key vote: {:?}",
+                                        txid
+                                    )
+                                }
+                                Err(e) => {
+                                    warn!("Failed to cast aggregate public key vote: {:?}", e);
+                                }
+                            }
                         }
                         EpochId::Epoch30 => {
-                            debug!("Received a DKG result, but are in epoch 3. Broadcast the transaction to stackerDB.");
-                            let signer_message =
-                                SignerMessage::Transactions(self.transactions.clone());
-                            if let Err(e) = self.stackerdb.send_message_with_retry(signer_message) {
-                                warn!("Failed to update transactions in stacker-db: {:?}", e);
+                            // TODO: get the latest round
```

Comment:
I think you have a function that does this now?

Reply:
Yep. Just haven't bothered using yet as I wanted to test some things first but have been blocked for over  aweek